src/green/transform.jl

```diff
@@ -134,54 +46,137 @@ end
 end
 
 #TODO: we need a version with the type of mesh as the second argument
-@generated function _find_mesh(meshs::MT, mesh::M) where {MT,M}
-    #find the first M type mesh in obj.mesh, if not found, return 0
-    # Core.println(MT, ", ", M)
+@generated function _find_mesh(::Type{MT}, ::Type{M}) where {MT,M}
     for (i, t) in enumerate(fieldtypes(MT))
-
         # type equality is implemented as t<:M and M<:t, 
         # see https://discourse.julialang.org/t/how-to-test-type-equality/14144/6?u=mrbug
-        if t == M
+        if t <: M
             return :($i)
         end
     end
     return :(0)
 end
 
+@generated function _find_mesh(::Type{MT}, ::Type{M1}, ::Type{M2}) where {MT,M1,M2}
+    for (i, t) in enumerate(fieldtypes(MT))
+        if t <: M1 || t <: M2
+            return :($i)
+        end
+    end
+    return :(0)
+end
+
+@generated function _find_mesh(::Type{MT}, ::Type{M1}, ::Type{M2}, ::Type{M3}) where {MT,M1,M2,M3}
+    for (i, t) in enumerate(fieldtypes(MT))
+        if t <: M1 || t <: M2 || t <: M3
+            return :($i)
+        end
+    end
+    return :(0)
+end
+
+"""
+    Base.:<<(objL::MeshArray, objR::MeshArray)
+
+DLR Fourier transform of functions on the first temporal grid (ImTime, ImFreq or DLRFreq). 
+
+- If objL and objR have identical temporal grid, objL<<objR assign objR to objL.
+- If objL and objR have different temporal grid, one of them has to be in DLR space.
+    * If objL is in DLR space, objL<<objR calculates the DLR spectral density of data in objR
+    * if objR is in DLR space, objL<<objR calculates the Green's function from the DLR spectral density in objR.
+"""
+function Base.:<<(objL::MeshArray{T,N,MT1}, objR::MeshArray{T,N,MT2}) where {T,N,MT1,MT2}
+    dimL = _find_mesh(MT1, ImTime, ImFreq, DLRFreq)
+    dimR = _find_mesh(MT2, ImTime, ImFreq, DLRFreq)
+    @assert dimL == dimR "The temporal dimensions should be identical for source and target MeshArrays."
+
+    typeL = typeof(objL.mesh[dimL])
+    typeR = typeof(objR.mesh[dimR])
+    meshL = objL.mesh[dimL]
+    meshR = objR.mesh[dimR]
+
+    if typeL == typeR
```

Comment:
this long if else structure reminds me that maybe it's better to implement dispatch here?
e.g. all xxx2xxx be implemented as transform(data, meshL::XXXMesh, meshR::YYYMesh), which performs transform from XXX to YYY

Reply:
while current to_xxx functions could be preserved for user's convenience and implemented by calling the proper transform function